services/src/datasets/storage.rs

```diff
@@ -159,6 +179,20 @@ pub trait DataSetDB: DataSetProvider + Send + Sync {
     ) -> Result<&dyn DataSetProvider>;
 }
 
+pub trait DataSetStorer: Send + Sync {
+    type D: Send + Sync;
+}
```

Comment:
if `D` is the meta data type, maybe you should call it
```suggestion
pub trait DataSetStorer: Send + Sync {
    type MetaData: Send + Sync;
}
```

Reply:
a0c4a7488f489f31a35aab8b045816f567874a98